src/core/metrics.cc

```diff
@@ -229,9 +228,20 @@ bool metric_id::operator==(
 // Unfortunately, metrics_impl can not be shared because it
 // need to be available before the first users (reactor) will call it
 
-shared_ptr<impl>  get_local_impl() {
-    static thread_local auto the_impl = ::seastar::make_shared<impl>();
-    return the_impl;
+shared_ptr<impl> get_local_impl(int handle) {
+    auto& impls = get_metric_implementations();
+
+    auto impl_iter = std::find_if(impls.begin(), impls.end(), [handle](const auto& crnt) {
+        return crnt->get_handle() == handle;
+    });
```

Comment:
Would it be preferable to use perhaps a `std::unordered_map<handle, shared_impl>` to be less intrusive to the existing API and avoid the linear search via a pointer?

Reply:
Good point. I don't see why we shouldn't do that.